yqlplus_engine/src/test/java/com/yahoo/yqlplus/engine/java/JavaProgramCompilerTest.java

```diff
@@ -3852,6 +3852,96 @@ protected void configure() {
         Assert.assertEquals(list.size(), 1);
     }
     
+    @Test
+    public void testSimpleLeftJoinOnCondition() throws Exception {
+        Injector injector = Guice.createInjector(new JavaTestModule(),
+                new SourceBindingModule("source", new IntSource(), "stringUtilUDF", StringUtilUDF.class)
```

Comment:
Does StringUtilUDF need to be registered for this test?

Reply:
This one doesn't need it. Removing it.